### PR TITLE
Upload coverage artifacts and configure sonarcloud

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/compound/issues/79

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+sonar.projectKey=compound-web
```
Comment (t3chguy) [inline sonar-project.properties:1]:
```suggestion
sonar.projectKey=vector-im_compound-web
```

Comment (t3chguy) [reply sonar-project.properties:1]: image

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+sonar.organization=vector-im
+
+
+# sonar.sources=src,res
+# sonar.tests=test,cypress
+sonar.exclusions=__test__,docs
+
+sonar.typescript.tsconfigPath=./tsconfig.json
+sonar.javascript.lcov.reportPaths=coverage/lcov.info
```
Comment (t3chguy) [inline sonar-project.properties:10]: this seems unused given the above coverage artifact is uploaded but never downloaded

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -26,5 +26,17 @@ jobs:
...
+              with:
+                  name: coverage
+                  path: |
+                      coverage
+                      !coverage/lcov-report
+            - name: SonarCloud Scan
+              uses: SonarSource/sonarcloud-github-action@master
+              env:
+                GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                SONAR_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
```
Comment (t3chguy) [inline .github/workflows/pull_request.yml:42]: This won't work for forked PRs, thus will mean you can't use coverage as a branch protection

Comment (t3chguy) [reply .github/workflows/pull_request.yml:42]: https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.

Comment (gsouquet) [reply .github/workflows/pull_request.yml:42]: Makes sense! I'm happy to punt that for now as we don't expect contributions for now.
But I have tracked it as part of https://github.com/vector-im/compound/issues/80

**Review by t3chguy — APPROVED** [review]

**Pull request merged**